addons/stock/models/stock_picking.py

```diff
@@ -929,6 +944,12 @@ def button_validate(self):
         pickings_to_backorder.with_context(cancel_backorder=False)._action_done()
         return True
 
+    def action_set_done_quantities(self):
+        precision_digits = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        for move in self.move_lines:
+            if move.product_id.tracking == 'none' and float_is_zero(move.quantity_done, precision_digits=precision_digits):
```

Comment:
I don't think we need this check again since it was already checked in the compute that display the action.